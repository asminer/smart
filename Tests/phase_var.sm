
// $Id: phase_avg.sm 1359 2012-12-21 21:49:58Z asminer $

// # Debug + procgen
// # Debug + exact_solver

real square(real x) := x*x;

print("\n----------------------------------------------------------------\n");
ph int zero := 0;
print("Variance for const(0):\n");
print("\tcomputed: ", var(zero), "\n");
print("\t  theory: ", 0.0, "\n");

print("\n----------------------------------------------------------------\n");
ph int one := 1;
print("Variance for const(1):\n");
print("\tcomputed: ", var(one), "\n");
print("\t  theory: ", 0.0, "\n");

print("\n----------------------------------------------------------------\n");
ph int inf := infinity;
print("Variance for const(oo):\n");
print("\tcomputed: ", var(inf), "\n");
print("\t  theory: ", 0.0, "\n");

print("\n----------------------------------------------------------------\n");
ph int die := equilikely(1,6);
print("Variance for equilikely(1,6):\n");
print("\tcomputed: ", var(die), "\n");
print("\t  theory: ", 35.0/12.0, "\n");

print("\n----------------------------------------------------------------\n");
ph int dice := equilikely(1,6) + equilikely(1,6);
print("Variance for dice:\n");
print("\tcomputed: ", var(dice), "\n");
print("\t  theory: ", 35.0/6.0, "\n");

print("\n----------------------------------------------------------------\n");
ph int coin := bernoulli(0.6);
print("Variance for bernoulli(0.6):\n");
print("\tcomputed: ", var(coin), "\n");
print("\t  theory: ", 0.4*0.6, "\n");

print("\n----------------------------------------------------------------\n");
ph int howmany := geometric(0.6);
print("Variance for geometric(0.6):\n");
print("\tcomputed: ", var(howmany), "\n");
print("\t  theory: ", 0.6 / square(1-0.6), "\n");

print("\n----------------------------------------------------------------\n");
ph int pascal := geometric(0.6) + geometric(0.6) + geometric(0.6);
print("Variance for pascal(3, 0.6):\n");
print("\tcomputed: ", var(pascal), "\n");
print("\t  theory: ", 3*0.6 / square(1-0.6), "\n");

print("\n----------------------------------------------------------------\n");
ph int coins := binomial(3, 0.6);
print("Variance for binomial(3, 0.6):\n");
print("\tcomputed: ", var(coins), "\n");
print("\t  theory: ", 3 * 0.6 * 0.4, "\n");

print("\n----------------------------------------------------------------\n");
ph int altcoins := bernoulli(0.6) + bernoulli(0.6) + bernoulli(0.6);
print("Variance for sum of 3 bernoulli's:\n");
print("\tcomputed: ", var(altcoins), "\n");
print("\t  theory: ", 3 * 0.6 * 0.4, "\n");

print("\n----------------------------------------------------------------\n");
print("Variance for expo(5):\n");
print("\tcomputed: ", var(expo(5)), "\n");
print("\t  theory: ", square(1.0/5.0), "\n");

print("\n----------------------------------------------------------------\n");
ph real x := expo(5);
print("Variance for ph real expo(5):\n");
print("\tcomputed: ", var(x), "\n");
print("\t  theory: ", square(1.0/5.0), "\n");

print("\n----------------------------------------------------------------\n");
ph real earl := erlang(3, 5);
print("Variance for erlang(3, 5):\n");
print("\tcomputed: ", var(earl), "\n");
print("\t  theory: ", 3.0*square(1.0/5.0), "\n");


print("\n----------------------------------------------------------------\n");
ph int flip := choose(0:1, infinity:1);
print("Variance of flip:\n");
print("\tcomputed: ", var(flip), "\n");
print("\t  theory: ", infinity, "\n");

print("\n");
