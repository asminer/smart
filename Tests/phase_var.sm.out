
----------------------------------------------------------------
Variance for const(0):
	computed: 0
	  theory: 0

----------------------------------------------------------------
Variance for const(1):
	computed: 0
	  theory: 0

----------------------------------------------------------------
Variance for const(oo):
	computed: 0
	  theory: 0

----------------------------------------------------------------
Variance for equilikely(1,6):
	computed: 2.91667
	  theory: 2.91667

----------------------------------------------------------------
Variance for dice:
	computed: 5.83333
	  theory: 5.83333

----------------------------------------------------------------
Variance for bernoulli(0.6):
	computed: 0.24
	  theory: 0.24

----------------------------------------------------------------
Variance for geometric(0.6):
	computed: 3.75
	  theory: 3.75

----------------------------------------------------------------
Variance for pascal(3, 0.6):
	computed: 11.25
	  theory: 11.25

----------------------------------------------------------------
Variance for binomial(3, 0.6):
	computed: 0.72
	  theory: 0.72

----------------------------------------------------------------
Variance for sum of 3 bernoulli's:
	computed: 0.72
	  theory: 0.72

----------------------------------------------------------------
Variance for expo(5):
	computed: 0.04
	  theory: 0.04

----------------------------------------------------------------
Variance for ph real expo(5):
	computed: 0.04
	  theory: 0.04

----------------------------------------------------------------
Variance for erlang(3, 5):
	computed: 0.12
	  theory: 0.12

----------------------------------------------------------------
Variance of flip:
	computed: infinity
	  theory: infinity

